Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-page-titles.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { createContext, useCallback, useContext, useMemo } from 'react';
+
+import { IHaveChildrenProps } from '../components/common/Props';
+
+interface IPageTitleParamsContext {
+    actions: {
+        setPageTitle: (title: string | undefined) => void;
+    };
+}
+
+type IPageTitleProviderProps = IHaveChildrenProps
+
+const PageTitleContext = createContext<IPageTitleParamsContext>({} as IPageTitleParamsContext);
+
+const PageTitleProvider = ({ children }: IPageTitleProviderProps) => {
+    const defaultTitlePage = useMemo(
+        () => 'SoftUni Judge',
+        [],
+    );
+
+    const setPageTitle = useCallback(
+        (title: string | undefined) => {
+            if (title === defaultTitlePage) {
+                document.title = defaultTitlePage;
+            } else {
+                document.title = `${title} - ${defaultTitlePage}`;
+            }
```

Comment:
What happends when `title` is nil (undefined or null)? As I understand it, it will become ` - SoftUni Judge`.
The check should be:
```
document.title = isNil(title)
    ? defaultPageTitle
    : `${title} - {defaultPageTitle}`
```